### PR TITLE
fix(text-field): update isRtl to a prop

### PR DESCRIPTION
**Pull request by moog16**
fixes https://github.com/material-components/material-components-web-react/issues/194

BREAKING CHANGE: In order for rtl to work on the Notched Outline component, isRtl will need to be passed in as boolean (default is false). Updating isRtl will result in a re-render.

### CODE REVIEW

File: test/unit/text-field/index.test.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ test('#componentDidUpdate calls setValue if state.value updates', () => {
 test('#componentDidUpdate does not call setValue if another property updates', () => {
   const wrapper = shallow(<TextField label='my label'><Input /></TextField>);
   wrapper.instance().foundation_.setValue = td.func();
-  wrapper.setState({dir: 'rtl'});
+  wrapper.setState({dir: 'rtl'}); // could be anything
```
Comment (bonniezhou) [inline test/unit/text-field/index.test.js]: What does "could be anything" mean?

File: test/screenshot/webpack.config.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
 module.exports = [
```
Comment (bonniezhou) [inline test/screenshot/webpack.config.js]: Uncomment these

File: test/unit/text-field/index.test.js [hunk 3]
```diff
@@ -107,24 +107,22 @@ test('#adapter.removeClass removes class from state.classList', () => {
...
-test('#adapter.isRtl returns true if the direction is true', () => {
-  const div = document.createElement('div');
-  // needs to be attached to real DOM to get width
-  // https://github.com/airbnb/enzyme/issues/1525
-  document.body.style.direction = 'rtl';
-  document.body.append(div);
-  const options = {attachTo: div};
-  const wrapper = mount(
-    <TextField label='my label'><Input /></TextField>, options);
+test('#adapter.isRtl returns true props.isRtl is true', () => {
```
Comment (bonniezhou) [inline test/unit/text-field/index.test.js]: forgot a word: "if"

File: test/unit/text-field/index.test.js [hunk 4]
```diff
@@ -107,24 +107,22 @@ test('#adapter.removeClass removes class from state.classList', () => {
...
+  const wrapper = shallow(
+    <TextField isRtl label='my label'><Input /></TextField>);
   assert.isTrue(wrapper.instance().foundation_.adapter_.isRtl());
-  document.body.style.direction = 'initial';
-  div.remove();
+});
+
+test('#adapter.isRtl returns false props.isRtl is false', () => {
```
Comment (bonniezhou) [inline test/unit/text-field/index.test.js]: forgot a word: "if"

File: packages/text-field/index.js [hunk 5]
```diff
@@ -184,13 +184,6 @@ class TextField extends React.Component {
     };
   }
 
-  getIsRtl = () => {
-    if (this.textFieldElement.current) {
```
Comment (bonniezhou) [inline packages/text-field/index.js]: Can we get rid of the ref to `textFieldElement` now?

File: packages/text-field/README.md [hunk 6]
```diff
@@ -62,6 +62,7 @@ dense | Boolean | Enables dense variant.
 floatingLabelClassName | String | An optional class added to the floating label element fo the text field.
 fullWidth | Boolean | Enables fullWidth variant.
 helperText | Element | Helper text that appears below the text field. Use the `<HelperText>` component that comes with this package.
+isRtl | Boolean | Updates the `<NotchedOutline>` to the correct orientation (left or right).
```
Comment (bonniezhou) [inline packages/text-field/README.md]: This description is an implementation detail, maybe say instead "Whether the direction of the text field element is set to RTL."

File: test/unit/text-field/index.test.js (hunk 1 above)
Comment (moog16) [reply test/unit/text-field/index.test.js]: I'll rewrite as  // could be any property/value pair. Or should i just remove the comment altogether?

**Pull request merged**